google_forms/reader.py

```diff
@@ -0,0 +1,80 @@
+from datadog import statsd
+from datetime import datetime
+import gspread
+from typing import Any, List, Optional
+
+from clients.laravel_client import bulk_upload_submissions
+from common.cache import LocalCache, S3Cache
+from common.config import (
+    GOOGLE_DATETIME_FORMAT,
+    GOOGLE_LAST_RUN_FILENAME,
+    TWITTER_LAST_RUN_BUCKET,
+    GOOGLE_DATETIME_FORMAT,
+    GOOGLE_FORMS_LARAVEL_API_KEY,
+)
+from common.data_classes import RawSubmission
+from common.enums import DataSource
+
+
+def run_form_reader(
+    account_creds_file: str,
+    form_id: str,
+    job_mode: str,
+    last_run_date_str: Optional[str] = None,
+    local_store: bool = True,
+) -> None:
+
+    curtime = datetime.now()
+
+    cache = LocalCache(GOOGLE_LAST_RUN_FILENAME)
+    if not local_store:
+        cache = S3Cache(GOOGLE_LAST_RUN_FILENAME, TWITTER_LAST_RUN_BUCKET)
+
+    if not last_run_date_str:
+        last_run_date_str = cache.get_last_processed_time_stamp_from_file()
+
+    last_run_datetime = datetime.strptime(last_run_date_str, GOOGLE_DATETIME_FORMAT)
+
+    gc = gspread.service_account(filename=account_creds_file)
+    sh = gc.open_by_key(form_id)
+    ws = sh.get_worksheet(0)
+
+    rows = ws.get_all_values()
+    statsd.increment(f"google_form.rows_processed", len(rows), tags=[f"job_mode:{job_mode}"])
+
+    results = _process_form_results(rows, last_run_datetime)
+    if results:
+        bulk_upload_submissions(results, GOOGLE_FORMS_LARAVEL_API_KEY, job_mode)
+
+    cache.log_last_processed_timestamp(curtime.strftime(GOOGLE_DATETIME_FORMAT))
+
+
+def _process_form_results(rows: List[List[Any]], last_run_datetime: datetime) -> List[RawSubmission]:
+    results = []
+    i = 0
+    for row in rows:
```

Comment:
Why nit use enumerate here? Then you can get rid of i

Reply:
Because I always forget enumerate exists